src/CoreWCF.BuildTools/src/OperationParameterInjectionGenerator.cs

```diff
@@ -0,0 +1,235 @@
+");
+            builder.Append($@"            System.IServiceProvider serviceProvider = CoreWCF.OperationContext.Current.InstanceContext.Extensions.Find<System.IServiceProvider>();
+            if (serviceProvider == null) throw new System.InvalidOperationException(""Missing IServiceProvider in InstanceContext extensions"");
+");
+            if (isInstanceContextModeSingle)
+            {
+                builder.Append($@"            using (var scope = serviceProvider.CreateScope())
+            {{
+");
+            }    
+            
+            AddDependenciesResolution();
+            AddMethodCall();
+
+            if (isInstanceContextModeSingle)
+            {
+                builder.Append($@"
+            }}");
+            }
+
+            builder.Append($@"
+        }}
+    }}
+}}
+");
+            context.AddSource(fileName, SourceText.From(builder.ToString(), Encoding.UTF8, SourceHashAlgorithm.Sha256));
+
+            string GetReturnType()
+                => operationContract.ReturnsVoid ?
+                    "void"
+                    : $"{operationContract.ReturnType.ContainingNamespace}.{operationContract.ReturnType.Name}";
+
+            string GetParameters()
+                => string.Join(", ", operationContract.Parameters
+                    .Select(p => $"{p.Type.ContainingNamespace}.{p.Type.Name} {p.Name}"));
+            
+            void AddDependenciesResolution()
+            {
+                string serviceProviderName = isInstanceContextModeSingle ?
+                    "scope"
+                    : "serviceProvider";
+                string prefix =  isInstanceContextModeSingle ?
+                    "                "
+                    : "            ";
+                for (int i = 0; i < dependencies.Length; i++)
+                {
+                    builder.AppendLine($@"{prefix}var d{i} = ({dependencies[i].Type.ContainingNamespace}.{dependencies[i].Type.Name}){serviceProviderName}.GetService(typeof({dependencies[i].Type.ContainingNamespace}.{dependencies[i].Type.Name}));");
+                }
+            }
+
+            void AddMethodCall()
+            {
+                string prefix = isInstanceContextModeSingle ?
+                 "                "
+                 : "            ";
+                var dependenciesParameters = Enumerable.Range(0, dependencies.Length).Select(x => $"d{x}");
+                builder.Append($"{prefix}return {operationContract.Name}({string.Join(", ", operationContract.Parameters.Select(x => x.Name).Union(dependenciesParameters))});");
+            }
+        }
+
+        private static IEnumerable<IMethodSymbol> FindOperationContractImplementationCandidate(GeneratorExecutionContext context, INamedTypeSymbol service, IMethodSymbol operationContract)
+        {
+            var CoreWCFInjectedAttribute = context.Compilation.GetTypeByMetadataName("CoreWCF.InjectedAttribute");
+            // Find a candidate implementation with same name as contract
```

Comment:
I liked the idea of using the ASP.NET Core FromServices attribute. But checking on that class, it involves pulling in the Mvc.Core assembly. Do you have any thoughts about the pros/cons? Should we have a similarly named attribute in a different namespace? Or should it be different name to prevent confusion/namespace ambiguity?

Reply:
I decided to introduce the `InjectedAttribute` for the the reasons you mentioned. First to avoid pulling the `Microsoft.AspNetCore.Mvc` package then to prevent confusion. Let me know if you think we should align the naming of the attribute with the one from ASP.NET Core. A good reason would be foreseeing a future `FrameworkDependency` to Microsoft.AspNetCore.App allowing to add compat code between CoreWCF and ASP.NET Core versions of `[FromServices]`. I do not know what your plan is taking a dependency on this for CoreWCF 2.0.